main.go

```diff
@@ -293,9 +308,53 @@ func checkParams(c *cli.Context) error {
 		return fmt.Errorf("Missing required param: cluster")
 	}
 
+	if !isValidKubectlVersionParam(c, extraKubectlVersions) {
+		return fmt.Errorf(getInvalidKubectlVersionMessage(c, extraKubectlVersions))
+	}
+
 	return nil
 }
 
+// isValidKubectlVersionParam tests whether a given version is valid within the current environment
+func isValidKubectlVersionParam(c *cli.Context, availableVersions []string) bool {
+	kubectlVersionParam := c.String("kubectl-version")
+	// using the default version
+	if kubectlVersionParam == "" {
+		return true
+	}
+
+	// using a custom version but no extra versions are available
+	if len(availableVersions) == 0 {
+		return false
+	}
+
+	// using a custom version ...
+	// return true if included in available extra versions; false otherwise
+	for _, availableVersion := range availableVersions {
+		if kubectlVersionParam == availableVersion {
+			return true
+		}
+	}
+	return false
+}
+
+// getInvalidKubectlVersionMessage returns an error message for invalid kubectl-version values
+func getInvalidKubectlVersionMessage(c *cli.Context, availableVersions []string) string {
+	kubectlVersionParam := c.String("kubectl-version")
+	// kubectl-version is valid; return an empty string
+	if isValidKubectlVersionParam(c, availableVersions) {
```

Comment:
This function assumes the version is invalid, so this is already checked before, i don't think it makes sense to check it again, otherwise the error message will have a null string